readium/streamer/src/main/java/org/readium/r2/streamer/parser/audio/AudioParser.kt

```diff
@@ -37,28 +45,46 @@ public class AudioParser : PublicationParser {
 
         val readingOrder =
             if (asset.mediaType.matches(MediaType.ZAB)) {
-                (asset.container.entries() ?: emptySet())
-                    .filter { entry -> zabCanContain(entry.url) }
-                    .sortedBy { it.url.toString() }
-                    .toMutableList()
+                asset.container
+                    .filter { zabCanContain(it) }
+                    .sortedBy { it.toString() }
```

Comment:
We don't retrieve the media type of each archive entry... And an archive typed ZAB could contain playlist files that should not be put into the `readingOrder`.

Reply:
I'm suggesting to sniff the proper media type of all the entries in the archive, and then filter them to construct the `readingOrder` using `isAudio` instead of the hard-coded list of audio extensions.